service/service.go

```diff
@@ -0,0 +1,98 @@
+package service
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+
+	btcecdsa "github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+)
+
+// Text used to signify that a signed message follows and to prevent
+// inadvertently signing a transaction.
+const messageSignatureHeader = "Bitcoin Signed Message:\n"
+
+var ErrInvalidDescriptor = errors.New("invalid descriptor: input is malformed")
+
+type R2PService struct {
+	router   *gin.Engine
+	pmClient IPlanetmintClient
+	eClient  IElementsClient
+}
+
+func NewR2PService(router *gin.Engine, pmClient IPlanetmintClient, eClient IElementsClient) *R2PService {
+	service := &R2PService{router: router, pmClient: pmClient, eClient: eClient}
+	service.configureRouter()
+	service.registerRoutes()
+	return service
+}
+
+func (r2p *R2PService) Run(config *viper.Viper) {
+	bindAddress := config.GetString("service-bind")
```

Comment:
this string got renamed in the this very PR in the config file. This resolution will thus always fail.